src/BPEssentials/ExtensionMethods/PlayerWarns.cs

```diff
@@ -13,45 +15,103 @@ public class SerializableWarn
         {
             public string Reason { get; set; }
 
+            public bool Expired { get; set; }
+
             public string IssueraccountID { get; set; }
 
-            public DateTime Date { get; set; }
+            public DateTimeOffset Date { get; set; }
+
+            public int Length { get; set; }
 
-            public SerializableWarn(string issuer, string reason, DateTime dateTime)
+            public SerializableWarn(string issuer, string reason, DateTimeOffset dateTime, int length = -1)
             {
                 IssueraccountID = issuer;
                 Reason = reason;
                 Date = dateTime;
+                if (length < 1)
+                {
+                    Length = length;
+                }
+                else
+                {
+                    Length = Core.Instance.Settings.Warns.DefaultWarnsExpirationInDays;
+                }
```

Comment:
Using a ternary operator would look a lot better in this use case.
```suggestion
                Length = length > 0 ? length : Core.Instance.Settings.Warns.DefaultWarnsExpirationInDays;
```